### PR TITLE
:sparkles: add FHIR resource validation endpoint that receives a POST…

### PR DESCRIPTION
**Pull request by YemreGurses**
… request to validate the FHIR resources using the onFHIR or FHIR validator based on a given redirectUrl parameter

Fixes: https://github.com/srdc/tofhir/issues/137

### CODE REVIEW

**Review by dogukan10 — CHANGES_REQUESTED** [review]
Can you add a test for this endpoint ? 
**Update: _It's added._**

File: tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala [hunk 1]
```diff
@@ -41,13 +42,42 @@ class FhirDefinitionsEndpoint(fhirDefinitionsConfig: FhirDefinitionsConfig) exte
...
+        }
+      }
+
+  /**
+   * Validates a FHIR resource using the onFHIR or FHIR validator (it is given as a parameter as redirectUrl)
+   * @return
+   */
+  private def validateResource(): Route = {
+    post {
+      parameters("redirectUrl") { redirectUrl =>
```
[the remark this answers is not in the document]
Comment (YemreGurses) [reply tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala:57]: Works for me

File: tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala [hunk 2]
```diff
@@ -41,13 +42,42 @@ class FhirDefinitionsEndpoint(fhirDefinitionsConfig: FhirDefinitionsConfig) exte
           }
         }
       }
+    } ~ pathPrefix(SEGMENT_VALIDATE) {
```
Comment (dogukan10) [inline tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala:45]: I think we can put it under pathPrefix(SEGMENT_FHIR_DEFINITIONS). WDYT @YemreGurses  ?

File: tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala [hunk 3]
```diff
@@ -41,13 +42,42 @@ class FhirDefinitionsEndpoint(fhirDefinitionsConfig: FhirDefinitionsConfig) exte
...
+        pathEndOrSingleSlash {
+          validateResource()
+        }
+      }
+
+  /**
+   * Validates a FHIR resource using the onFHIR or FHIR validator (it is given as a parameter as redirectUrl)
```
Comment (dogukan10) [inline tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala:52]: It should be **fhirValidationUrl**

File: tofhir-server/src/main/scala/io/tofhir/server/service/FhirDefinitionsService.scala [hunk 4]
```diff
@@ -108,4 +124,46 @@ class FhirDefinitionsService(fhirDefinitionsConfig: FhirDefinitionsConfig) {
     simplifiedStructureDefinitionCache.getOrElseUpdate(profileUrl, simpleStructureDefinitionService.simplifyStructureDefinition(profileUrl))
   }
 
+  /**
+   * Redirects the request to the onFHIR/FHIR validator based on the given redirectUrl and returns the response.
```
Comment (dogukan10) [inline tofhir-server/src/main/scala/io/tofhir/server/service/FhirDefinitionsService.scala:128]: It should be **fhirValidationUrl**

**Review by dogukan10 — CHANGES_REQUESTED** [review]

File: tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala (hunk 2 above)
Comment (dogukan10) [reply tofhir-server/src/main/scala/io/tofhir/server/endpoint/FhirDefinitionsEndpoint.scala:45]: We decided to leave it as it is

**Review by dogukan10 — APPROVED** [review]

**Pull request merged**